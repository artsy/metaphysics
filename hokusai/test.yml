version: "2"
services:
  metaphysics:
    command: env NODE_OPTIONS=--max_old_space_size=4096 yarn ci
    extends:
      file: build.yml
      service: metaphysics
    environment:
      - MEMCACHED_URL=metaphysics-memcached:11211
      - CI_PULL_REQUEST=$CI_PULL_REQUEST
      - CIRCLE_PULL_REQUEST=$CIRCLE_PULL_REQUEST
      - CIRCLE_BUILD_NUM=$CIRCLE_BUILD_NUM
      - ENABLE_CONSIGNMENTS_STITCHING=true
    env_file: ../.env.test
    depends_on:
      - metaphysics-memcached
  metaphysics-memcached:
    image: memcached
    ports:
      - 11211:11211
