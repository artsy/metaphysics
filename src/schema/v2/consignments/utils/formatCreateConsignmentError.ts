import {
  GraphQLInt,
  GraphQLNonNull,
  GraphQLObjectType,
  GraphQLString,
} from "graphql"
import { formatGravityError } from "lib/gravityErrorHandler"
import { ResolverContext } from "types/graphql"

export const ConsignmentInquiryMutationErrorName =
  "ConsignmentInquiryMutationError"

export const ConsignmentInquiryErrorType = new GraphQLObjectType<
  any,
  ResolverContext
>({
  name: ConsignmentInquiryMutationErrorName,
  fields: () => ({
    type: {
      type: GraphQLString,
    },
    message: {
      type: new GraphQLNonNull(GraphQLString),
    },
    error: {
      type: GraphQLString,
    },
    statusCode: {
      type: GraphQLInt,
    },
  }),
})

export const formatCreateConsignmentError = (error) => {
  return formatGravityError(error)
}
