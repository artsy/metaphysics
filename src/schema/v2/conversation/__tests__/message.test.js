/* eslint-disable promise/always-return */
import { runAuthenticatedQuery } from "schema/v2/test/utils"

describe("Me", () => {
  describe("Conversation", () => {
    describe("Message", () => {
      const context = {
        conversationLoader: () =>
          Promise.resolve({
            id: "420",
            initial_message: "Loved some of the works at your fair booth!",
            from_email: "collector@example.com",
          }),
        conversationMessagesLoader: () =>
          Promise.resolve({
            total_count: 2,
            message_details: [
              {
                id: "222",
                raw_text: "I'm a cat",
                is_first_message: true,
                from_email_address: "fancy_german_person@posteo.de",
                from_id: null,
                attachments: [],
                metadata: {
                  lewitt_invoice_id: "420i",
                },
                from: `"Percy Z" <percy@cat.com>`,
                from_principal: true,
                original_text:
                  "Loved some of the works at your fair booth! ABOUT THIS COLLECTOR: blah, blah",
                body: "I'm a cat",
                to: [],
                cc: [],
              },
              {
                id: "222",
                raw_text: "I'm a cat",
                is_first_message: false,
                from_email_address: "fancy_german_person@posteo.de",
                from_id: null,
                attachments: [],
                metadata: {
                  lewitt_invoice_id: "420i",
                },
                from: `"Percy Z" <percy@cat.com>`,
                from_principal: true,
                original_text: "I'm a cat oh yea!",
                body: "I'm a cat",
                to: ['"Recipient Name" <recipient@cat.com>'],
                cc: ['"Another Recipient" <another_recipient_in_cc@cat.com>'],
              },
            ],
          }),
      }

      it("returns sanitized messages", () => {
        const query = `
          {
            me {
              conversation(id: "420") {
                internalID
                initialMessage
                from {
                  email
                }
                messagesConnection(first: 10, includeDeliveryPending: true) {
                  totalCount
                  edges {
                    node {
                      isFirstMessage
                      body
                      internalID
                      from {
                        email
                        name
                      }
                      to
                      cc
                    }
                  }
                }
              }
            }
          }
        `

        return runAuthenticatedQuery(query, context).then(
          ({ me: conversation }) => {
            expect(conversation).toMatchSnapshot()
          }
        )
      })

      it("handles null message bodies", () => {
        const query = `
          {
            me {
              conversation(id: "420") {
                internalID
                initialMessage
                from {
                  email
                }
                messages(first: 10) {
                  edges {
                    node {
                      body
                      internalID
                    }
                  }
                }
              }
            }
          }
        `

        const message = {
          message_details: [
            {
              body: "null",
              id: "222",
            },
          ],
        }

        const customRootValue = Object.assign({}, context, {
          conversationMessagesLoader: () => Promise.resolve(message),
        })

        return runAuthenticatedQuery(query, customRootValue).then(
          ({ me: { conversation } }) => {
            expect(conversation).toMatchSnapshot()
          }
        )
      })

      it("returns proper is_from_user", () => {
        const query = `
          {
            me {
              conversation(id: "420") {
                internalID
                initialMessage
                from {
                  email
                }
                messages(first: 10) {
                  edges {
                    node {
                      body
                      internalID
                      isFromUser
                      from {
                        email
                        name
                      }
                    }
                  }
                }
              }
            }
          }
        `

        return runAuthenticatedQuery(query, context).then(
          ({ me: conversation }) => {
            expect(conversation).toMatchSnapshot()
          }
        )
      })
    })
  })
})
