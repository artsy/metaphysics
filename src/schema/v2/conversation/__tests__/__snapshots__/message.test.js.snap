// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Me Conversation Message handles null message bodies 1`] = `
{
  "from": {
    "email": "collector@example.com",
  },
  "initialMessage": "Loved some of the works at your fair booth!",
  "internalID": "420",
  "messages": {
    "edges": [],
  },
}
`;

exports[`Me Conversation Message returns proper is_from_user 1`] = `
{
  "conversation": {
    "from": {
      "email": "collector@example.com",
    },
    "initialMessage": "Loved some of the works at your fair booth!",
    "internalID": "420",
    "messages": {
      "edges": [
        {
          "node": {
            "body": "Loved some of the works at your fair booth!",
            "from": {
              "email": "fancy_german_person@posteo.de",
              "name": "Percy Z",
            },
            "internalID": "222",
            "isFromUser": true,
          },
        },
        {
          "node": {
            "body": "I'm a cat oh yea!",
            "from": {
              "email": "fancy_german_person@posteo.de",
              "name": "Percy Z",
            },
            "internalID": "222",
            "isFromUser": true,
          },
        },
      ],
    },
  },
}
`;

exports[`Me Conversation Message returns sanitized messages 1`] = `
{
  "conversation": {
    "from": {
      "email": "collector@example.com",
    },
    "initialMessage": "Loved some of the works at your fair booth!",
    "internalID": "420",
    "messagesConnection": {
      "edges": [
        {
          "node": {
            "body": "Loved some of the works at your fair booth!",
            "cc": [],
            "from": {
              "email": "fancy_german_person@posteo.de",
              "name": "Percy Z",
            },
            "internalID": "222",
            "isFirstMessage": true,
            "to": [],
          },
        },
        {
          "node": {
            "body": "I'm a cat oh yea!",
            "cc": [
              ""Another Recipient" <another_recipient_in_cc@cat.com>",
            ],
            "from": {
              "email": "fancy_german_person@posteo.de",
              "name": "Percy Z",
            },
            "internalID": "222",
            "isFirstMessage": false,
            "to": [
              ""Recipient Name" <recipient@cat.com>",
            ],
          },
        },
      ],
      "totalCount": 2,
    },
  },
}
`;
